cvat-ui/src/actions/requests-async-actions.ts

```diff
@@ -0,0 +1,138 @@
+                    if (target === 'task') {
+                        instance = { id: taskID as number, type: target, requestPromise: listeningPromise };
+                    } else if (target === 'job') {
+                        instance = { id: jobID as number, type: target, requestPromise: listeningPromise };
+                    } else if (target === 'project') {
+                        instance = { id: projectID as number, type: target, requestPromise: listeningPromise };
+                    }
+
+                    if (operationType === 'export') {
+                        if (operationTarget === 'backup') {
+                            if (!(state.export.tasks.backup.current?.[(instance as RequestInstanceType).id] ||
+                                state.export.projects.backup.current?.[(instance as RequestInstanceType).id])
+                            ) {
+                                dispatch(exportBackupAsync(
+                                    instance as RequestInstanceType,
+                                    undefined,
+                                    undefined,
+                                    undefined,
+                                    listeningPromise,
+                                ));
+                            }
+                        } else if (!(state.export.tasks.dataset.current?.[(instance as RequestInstanceType).id] ||
+                            state.export.projects.dataset.current?.[(instance as RequestInstanceType).id] ||
+                            state.export.jobs.dataset.current?.[(instance as RequestInstanceType).id])
+                        ) {
+                            dispatch(exportDatasetAsync(
+                                instance as RequestInstanceType,
+                                format,
+                                operationTarget === 'dataset',
+                                true,
+                                undefined,
+                                undefined,
+                                listeningPromise,
+                            ));
+                        }
+                    } else if (operationType === 'import') {
+                        if (!(state.import.tasks.dataset.current?.[(instance as RequestInstanceType).id] ||
+                            state.import.projects.dataset.current?.[(instance as RequestInstanceType).id] ||
+                            state.import.jobs.dataset.current?.[(instance as RequestInstanceType).id])
+                        ) {
+                            dispatch(importDatasetAsync(
+                                instance as RequestInstanceType,
+                                format,
+                                undefined,
+                                undefined,
+                                undefined,
+                                undefined,
+                                listeningPromise,
+                            ));
```

Comment:
May you please explain logic behind this? 

It makes signature and implemenetion of `exportBackupAsync`, `exportDatasetAsync`, `importDatasetAsync` more difficult, but I can't see the reason.

Reply:
Reworked as discussed, using separate listening functions